app/components/optics/button/component.rb

```diff
@@ -11,7 +11,9 @@ class Component < ApplicationViewComponent
       }
 
       accepts :label
```

Comment:
Is label necessary, or could this be built to always take nested content?

Also don't think we need to leading icon slot since what we talked about would mean everything comes through content.

Reply:
Also the options this should accept are:

- variant ✅
- border ✅
- disabled ✅
- active ✅
- pill
- icon ✅
- icon-with-label
- size ✅

Reply:
Is there a way to allow passing any other arguments passed in onto the button? Is that what the **@attributes does?

Could see if someone wants to set a test id or some other attribute, we don't want to  restrict things too much

Reply:
> Is label necessary, or could this be built to always take nested content?
> 
> Also don't think we need to leading icon slot since what we talked about would mean everything comes through content.

`accepts` are optional. Without a `label` the content block is rendered. `label` is more compact:
```slim
  = component 'optics/button', label: 'Save'
```
vs
```slim
  = component 'optics/button' do
    | Save
```

I used leading icon in Ephemeris but it should be easy enough to refactor.

Reply:
> Is there a way to allow passing any other arguments passed in onto the button? Is that what the **@attributes does?
> 
> Could see if someone wants to set a test id or some other attribute, we don't want to restrict things too much

Yes, you can pass arbitrary attributes.